pints/_mcmc/_relativistic.py

```diff
@@ -158,6 +161,68 @@ def ask(self):
         self._ready_for_tell = True
         return np.array(self._position, copy=True)
 
+    def _calculate_momentum_distribution(self):
+        """Calculate an approximation to the CDF of momentum magnitude.
+
+        The purpose of this method is to calculate self._inv_cdf, a function
+        giving a numerical approximation to the inverse cumulative distribution
+        function of the magnitude of the momentum vector. This function can be
+        used to perform inverse transform sampling to generate samples of the
+        momentum.
+
+        This method should be called before any sampling is performed. As
+        implemented, it is called upon the first call to self.ask.
+
+        Numerical integration of the momentum density is performed using the
+        trapezoidal rule.
+        """
+        num_adaptations = 0
+
+        # Set initial values for the integration grid
+        max_value = 10
+        spacing = 1e-4
+
+        integration_accepted = False
+
+        while not integration_accepted:
+            # Evaluate the logpdf on a grid
+            integration_grid = np.arange(1e-6, max_value, spacing)
+            logpdf_values = self._momentum_logpdf(integration_grid)
+
+            # Integrate using the trapezoidal rule
```

Comment:
Maybe say that we don't need to normalise it with the grid space because we will 're-normalise' all in one for the CDF? Just in case someone looks at it later and thinking why we didn't do the full trapezoidal rule.